src/components/Users/UserAvatar.tsx

```diff
@@ -66,14 +65,17 @@ export default function UserAvatar({ username }: { username: string }) {
   };
 
   const handleAvatarDelete = async (onError: () => void) => {
-    const { res } = await request(routes.deleteProfilePicture, {
-      pathParams: { username },
-    });
-    if (res?.ok) {
-      queryClient.invalidateQueries({ queryKey: ["currentUser"] });
-      toast.success(t("profile_picture_deleted"));
-      setEditAvatar(false);
-    } else {
+    try {
+      const response = await callApi(routes.deleteProfilePicture, {
+        pathParams: { username },
+      });
+
+      if (response) {
+        queryClient.invalidateQueries({ queryKey: ["currentUser"] });
+        toast.success(t("profile_picture_deleted"));
+        setEditAvatar(false);
+      }
+    } catch (_error) {
```

Comment:
same here, use `mutate`. Do not `try catch` things. you can use `onError` and `onSuccess` for callbacks if necessary.